casper/src/test/scala/coop/rchain/casper/api/BondedStatusAPITest.scala

```diff
@@ -0,0 +1,57 @@
+package coop.rchain.casper.api
+
+import cats.implicits._
+import com.google.protobuf.ByteString
+import coop.rchain.casper.engine.Engine
+import coop.rchain.casper.helper.TestNode.Effect
+import coop.rchain.casper.helper._
+import coop.rchain.casper.protocol._
+import coop.rchain.casper.util.GenesisBuilder._
+import coop.rchain.casper.EngineWithCasper
+import coop.rchain.crypto.PublicKey
+import coop.rchain.crypto.signatures.Secp256k1
+import coop.rchain.metrics.Metrics
+import coop.rchain.shared.scalatestcontrib._
+import coop.rchain.shared.Cell
+import monix.eval.Task
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.{EitherValues, FlatSpec, Matchers}
+
+class BondedStatusAPITest
+    extends FlatSpec
+    with Matchers
+    with EitherValues
+    with BlockGenerator
+    with BlockDagStorageFixture {
+  val genesisContext = buildGenesis()
+
+  implicit val metricsEff = new Metrics.MetricsNOP[Task]
+
+  def bondedStatus(publicKey: PublicKey)(node: TestNode[Effect]): Task[Boolean] = {
+    import node.logEff
+    val engine = new EngineWithCasper[Task](node.casperEff)
+    Cell.mvarCell[Task, Engine[Task]](engine).flatMap { implicit engineCell =>
+      BlockAPI
+        .bondStatus[Task](
+          BondStatusQuery(publicKey = ByteString.copyFrom(publicKey.bytes))
+        )
+        .map(_.right.value.isBonded)
+    }
+  }
+
+  "bondStatus" should "return true for bonded validator" in effectTest {
+    TestNode.networkEff(genesisContext, networkSize = 3).use {
+      case n1 +: n2 +: n3 +: _ =>
+        (bondedStatus(n1.validatorId.publicKey)(n1) shouldBeF true) >>
+          (bondedStatus(n2.validatorId.publicKey)(n1) shouldBeF true) >>
+          (bondedStatus(n3.validatorId.publicKey)(n1) shouldBeF true)
+    }
+  }
+
+  "bondStatus" should "return false for not bonded validators" in effectTest {
+    TestNode.standaloneEff(genesisContext).use { node =>
+      val (_, publicKey) = Secp256k1.newKeyPair
+      bondedStatus(publicKey)(node) shouldBeF false
+    }
+  }
+}
```

Comment:
We are missing a test showing that bondStatus returns false for a validator that is bonded in a block which is not yet finalized